xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-webjar/src/main/webjar/lightbox.js

```diff
@@ -0,0 +1,241 @@
+    return new URL(imageLink).pathname.split("/").pop();
+  };
+
+  /**
+   * Rescale image to original size.
+   */
+  var removeResizeParams = function(imageSrc) {
+    var url = new URL(imageSrc);
+    var searchParams = new URLSearchParams(url.search);
+    searchParams.delete('width');
+    searchParams.delete('height');
+    url.search = searchParams.toString();
+    return url.toString();
+  };
+
+  /**
+   * Scale image to thumbnail. This may not be applied for external urls.
+   */
+  var createThumbnailURL = function(imageSrc) {
+    var url = new URL(imageSrc);
+    var searchParams = new URLSearchParams(url.search);
+    searchParams.append('width', '150');
+    searchParams.append('height', '150');
+    url.search = searchParams.toString();
+    return url.toString();
+  };
+
+  /**
+   * Open lightbox in fullscreen mode, or close it if already open.
+   */
+  var toggleLightboxFullscreen = function() {
+    if (!$('#lightboxFullscreen').data('open')) {
+      myOpenLightbox.requestFullScreen($('#blueimp-gallery')[0]);
+      $('#lightboxFullscreen').data('open', true);
+    } else {
+      myOpenLightbox.exitFullScreen();
+      $('#lightboxFullscreen').data('open', false);
+    }
+  };
+
+  /**
+   * Update lightbox description using given information.
+   */
+  var updateLightboxDescription = function(data, caption) {
+    if (data != undefined) {
+      $('.lightboxDescription').css('display', 'block');
+      $('.lightboxDescription').find('.title').text(data.name);
+      var translationPublisher = l10n.get('author', XWiki.Model.resolve(data.author, XWiki.EntityType.DOCUMENT).name);
+      $('.lightboxDescription').find('.publisher').text(translationPublisher);
+      var translationDate = l10n.get('date', new Date(data.date).toLocaleDateString());
+      $('.lightboxDescription').find('.date').text(translationDate);
+
+      if (caption != undefined) {
+        $('.lightboxDescription').find('.caption').text(caption);
+      } else {
+        $('.lightboxDescription').find('.caption').empty();
+      }
+    } else {
+      $('.lightboxDescription').css('display', 'none');
+    }
+  };
+
+  /**
+   * Get information for attachment, when this can be found inside the current page.
+   * TODO: consider the path of the attachment, not just the current page.
+   */
+  var getAttachmentInfo = function(imageLink) {
+    var deferred = $.Deferred();
+    var attachmentsURL = xm.restURL + `/attachments`;
+    $.ajax(attachmentsURL, {
+      method: 'GET',
+      dataType: 'json'
+    }).done(function (data, textStatus, jqXHR) {
+      var attachment = data.attachments.find(field => field.name == getImageName(imageLink));
+      deferred.resolve(attachment);
+    }).fail(function () {
+      deferred.reject();
+    });
+    return deferred.promise();
+  };
+
+  var getImageCaption = function(image) {
+    if (image.closest('figure').length > 0) {
+      return image.closest('p').siblings('figcaption').text();
```

Comment:
This assume the `<img>`-tag is always nested in a paragraph. Unfortunately, this won't be the case with the new image syntax as from my tests then the image with the caption is not recognized by CKEditor. Therefore, it would be good if you could find the `<figcaption>` as a child of the `<figure>`-tag that you are already locating in this code.

Further, I'm wondering why you are only copying the text which looses all formatting and links instead of just cloning all children of the `<figcaption>`-element. For example if you have a link saying "click here", it is really strange if this is not a link in the lightbox. Note that there will (probably) be changes to include block-level content (and it is kind of already the case if you just press enter and start a second paragraph in the figure caption today) so make sure you use a `<div>` as wrapper and not a `<p>` if you copy/clone the children.